src/main/java/duke/Command.java

```diff
@@ -0,0 +1,295 @@
+     */
+    private void addEvent(String taskDetails, UI ui, TaskList tasks) throws DukeException {
+        if ((taskDetails == null) || !(taskDetails.contains(" /at "))) {
+            throw new DukeException("Incorrect Format of the Event Command!!, " +
+                    "Correct Format --> event <Description> /at <dd/MM/yyyy HHmm>");
+        } else {
+            String[] values = taskDetails.split(" /at ", 2);
+            try {
+                EventTask event = new EventTask(values[0], values[1]);
+                tasks.addTask(event);
+                ui.display("Got it. I've added this task:");
+                ui.display("  " + event);
+                ui.display("Now you have " + tasks.getTaskListLength() + " tasks in the list.");
+            } catch (DateTimeParseException e) {
+                throw new DukeException("Incorrect Format of the Event Command!!, " +
+                        "Correct Format --> event <Description> /at <dd/MM/yyyy HHmm>");
+            }
+        }
+    }
+
+    /**
+     * Method to check for the Task with the given String in the Task List
+     *
+     * @param searchDetails The String to be Searched in the Task List
+     * @param ui The Object to render the UI
+     * @param tasks The Object to contain the List of the Tasks
+     * @throws DukeException An Exception class to be thrown if the command taskDetails is not valid
+     */
+    private void searchAndDisplayTaskList(String searchDetails, UI ui, TaskList tasks) throws DukeException {
+        if (searchDetails != null && !searchDetails.equals("")) {
+            ui.printTaskList(tasks.searchTaskList(searchDetails));
+        } else {
+            throw new DukeException("OOPS! I'm sorry, but I don't know that command");
+        }
+    }
+    /**
+     * Method to check for the List Command and render the UI
+     *
+     * @param taskDetails The String describing the various attributes for the command
+     * @param ui The Object to render the UI
+     * @param tasks The Object to contain the List of the Tasks
+     * @throws DukeException An Exception class to be thrown if the command taskDetails is not valid
+     */
+    private void displayTaskList(String taskDetails, UI ui, TaskList tasks) throws DukeException {
+        if (taskDetails != null && taskDetails.equals("")) {
+            ui.printTaskList(tasks);
+        } else {
+            throw new DukeException("OOPS! I'm sorry, but I don't know that command");
+        }
+    }
+
+    /**
+     * Method to check for the ToDO Command and render the UI
+     *
+     * @param taskDetails The String describing the various attributes for the command
+     * @param ui The Object to render the UI
+     * @param tasks The Object to contain the List of the Tasks
+     * @throws DukeException An Exception class to be thrown if the command taskDetails is not valid
+     */
+    private void addTodo(String taskDetails, UI ui, TaskList tasks) throws DukeException {
```

Comment:
Same as the previous comment, taskDetail instead of taskDetails?

Reply:
Understood.